src/modules/deliveryoptimization/src/lib/DeliveryOptimization.c

```diff
@@ -0,0 +1,417 @@
+    {
+        JSON_Value* rootValue = json_parse_file(g_deliveryOptimizationConfigFile);
+
+        if (json_value_get_type(rootValue) == JSONObject)
+        {
+            JSON_Object* rootObject = json_value_get_object(rootValue);
+
+            if (json_object_has_value_of_type(rootObject, jsonPropertyName, jsonPropertyType))
+            {
+                JSON_Value* value = json_object_get_value(rootObject, jsonPropertyName);
+                char* json = json_serialize_to_string(value);
+
+                if (NULL != json)
+                {
+                    *payloadSizeBytes = strlen(json);
+                    if ((g_maxPayloadSizeBytes > 0) && ((unsigned)*payloadSizeBytes > g_maxPayloadSizeBytes))
+                    {
+                        OsConfigLogError(DeliveryOptimizationGetLog(), "MmiGet(%s, %s) insufficient maxmimum size (%d bytes) versus data size (%d bytes), reported value will be truncated", componentName, objectName, g_maxPayloadSizeBytes, *payloadSizeBytes);
+                        *payloadSizeBytes = g_maxPayloadSizeBytes;
+                    }
+
+                    *payload = (MMI_JSON_STRING)malloc(*payloadSizeBytes);
+                    if (NULL != *payload)
+                    {
+                        strncpy(*payload, json, *payloadSizeBytes);
+                    }
+                    else 
+                    {
+                        OsConfigLogError(DeliveryOptimizationGetLog(), "MmiGet failed to allocate %d bytes", *payloadSizeBytes + 1);
+                        *payloadSizeBytes = 0;
+                        status = ENOMEM;
+                    }
+
+                    json_free_serialized_string(json);
+                }
+                else 
+                {
+                    if (IsFullLoggingEnabled())
+                    {
+                        OsConfigLogError(DeliveryOptimizationGetLog(), "MmiGet failed to serialize JSON property (%s)", jsonPropertyName);
+                    }
+                    status = EINVAL;
+                }
+            }
+            else 
+            {
+                if (IsFullLoggingEnabled())
+                {
+                    OsConfigLogError(DeliveryOptimizationGetLog(), "MmiGet failed to find JSON property (%s)", jsonPropertyName);
+                }
+                status = EINVAL;
+            }
+        }
+        else 
+        {
+            if (IsFullLoggingEnabled())
+            {
+                OsConfigLogError(DeliveryOptimizationGetLog(), "MmiGet failed to parse JSON file (%s)", g_deliveryOptimizationConfigFile);
+            }
+            status = EINVAL;
+        }
+
+        if (NULL != rootValue)
+        {
+            json_value_free(rootValue);
+        }
+
+        // Reset status and payload if JSON file could not be parsed or property not found, as it may yet have to be configured.
+        if (MMI_OK != status)
+        {
+            const char* emptyJsonPayload = NULL;

```

Comment:
>const char* emptyJsonPayload = NULL; [](http://example.com/codeflow?start=12&length=36)

Another instance of many. Will stop commenting here.

Reply:
Thanks, fixed!